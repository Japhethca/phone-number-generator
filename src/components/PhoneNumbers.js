import React from 'react';
import { css } from 'glamor';


const phoneNumbersStyles = css({
  marginTop: '30px',
  overflowX: 'auto',
  height: '50vh',
  maxHeight: '60vh',
  backgroundColor: '#242124',
  columnCount: 4,
  columnFill: 'auto',
  padding: '10px',
  borderRadius: '5px',
});

const numberListStyles = css({
  marginTop: 0,
});

const numberStyles = css({
  padding: '2px',
  fontSize: '18',
});

const loaderStyles = css({
  fontSize: '14pt',
  fontWeight: 'bold',
});

function PhoneNumbers({numbers, isFetchingNumbers }) {
  return (
    <div {...phoneNumbersStyles}>
      { isFetchingNumbers 
        ? <div {...loaderStyles}> Loading...</div>
        : (
            <ol {...numberListStyles} id="copyNumbers">
              {numbers.map((number) => <li {...numberStyles} key={number}>{number}</li>)}
            </ol> 
          )
      }
    </div>
  );
}

export default PhoneNumbers;

