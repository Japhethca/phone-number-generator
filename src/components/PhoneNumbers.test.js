import React from 'react';
import renderer from 'react-test-renderer';

import PhoneNumbers from './PhoneNumbers';

describe('<PhoneNumbers />', () => {
  const props = {
    isFetchingNumbers: false,
    numbers: [
      '0032232152',
      '0951119083',
      '0987547892',
      '0909823787',
      '0232447837',
    ],
  };

  it('should render successfully', () => {
    const tree = renderer.create(<PhoneNumbers {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
