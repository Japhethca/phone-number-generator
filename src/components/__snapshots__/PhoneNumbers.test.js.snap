// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneNumbers /> should render successfully 1`] = `
<div
  data-css-g7q9rr=""
>
  <ol
    data-css-1j4x2ld=""
    id="copyNumbers"
  >
    <li
      data-css-r6gsmo=""
    >
      0032232152
    </li>
    <li
      data-css-r6gsmo=""
    >
      0951119083
    </li>
    <li
      data-css-r6gsmo=""
    >
      0987547892
    </li>
    <li
      data-css-r6gsmo=""
    >
      0909823787
    </li>
    <li
      data-css-r6gsmo=""
    >
      0232447837
    </li>
  </ol>
</div>
`;
